### ISSUE TITLE
Support for L colorspace and extra channels.

### ISSUE THREAD

**Issue by alexjc** [olokelo/jxlpy#14]
For the record in case people are looking: I have implemented both these features in my own branch at https://github.com/alexjc/jxlpy.

Just need to merge #13 first, then I'll port them over!

**Comment by olokelo** [comment]
@alexjc Thanks for your awesome work. Unfortunately I've run into an issue with Github actions package building and I have little time to fix it lately. It might also resolve the [#10 issue](https://github.com/olokelo/jxlpy/issues/10) building wheels for Windows and Mac OS. I would like to fix that first in order to make PyPI release with prebuilt wheels. If you have any ideas on how to manage building wheel packages for jxlpy in Github actions I would appreciate any help.

**Comment by alexjc** [comment]
Is there any more information than in #10?

**Comment by olokelo** [comment]
Not really. The current build pipeline is a bit hacky, it uses CIBW_BEFORE_ALL to build libjxl and then cibuildwheel to build and fix Python package. The libjxl build instructions are Linux specific and use some workarounds which don't work anymore with latest version of libjxl.
I had ran into issues trying to split build process into steps.

**Comment by alexjc** [comment]
How do you test it locally? I'll take a look.

**Comment by alexjc** [comment]
If you could merge #13 that would help, so I can get it working with the latest version.

**Comment by olokelo** [comment]
> How do you test it locally? I'll take a look.

I tested it with [Act](https://github.com/nektos/act). Thanks :)

> If you could merge #13 that would help, so I can get it working with the latest version.

Okay, will do.

**Comment by olokelo** [comment]
Awesome! Merged, Thanks again :)

I've also somewhat tested your master branch and it seems like [your change](https://github.com/alexjc/jxlpy/blob/d412eba90584269e1ae8ce1ed1c2f7bddbcf3bbe/_jxlpy/_jxl.pyx#L1118) in `get_frame` breaks printing pixel colors in examples. Could you please review it?

**Comment by alexjc** [comment]
Oh, yes I know what that is — relates to the extra channels. I will fix it...

**Comment by alexjc** [comment]
Unfortunately, my implementation of `L` and `I;16` modes from PIL doesn't work.  It seems to use a different stride somewhere and I don't know where it is...

I set `num_channels`, `num_color_channels`, and `bits_per_sample correctly` — something missing.  Advice welcome!

![displacement-as_8bit](https://user-images.githubusercontent.com/445208/223972746-69abf54b-7920-48ef-bbfe-8f4f8a4eebbb.jpg)

**Comment by alexjc** [comment]
OK, it's fixed it was a bug in ImageMagick actually.
